modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryTuplePrefix.java

```diff
@@ -51,13 +52,36 @@ public BinaryTuplePrefix(int elementCount, ByteBuffer buffer) {
         super(elementCount, buffer);
     }
 
+    /**
+     * Creates a prefix from provided {@link BinaryTuple}. If given tuple has lesser or equal number of
+     * columns, then all elements will be used in resulting prefix. If given tuple has more columns, then
+     * excess columns will be truncated.
+     *
+     * @param size The size of the complete tuple.
+     * @param tuple Tuple to create a prefix from.
+     * @return Prefix, created from provided tuple with regards to desired size.
+     */
+    public static BinaryTuplePrefix fromBinaryTuple(int size, BinaryTuple tuple) {
+        if (size == tuple.elementCount()) {
+            return entireTuple(tuple);
+        } else if (size > tuple.elementCount()) {
```

Comment:
I'm not sure what exactly you are asking...

as utility method, it must cover all possible inputs. In that particular case, size of the given BinaryTuple may be greater, lesser, or equal to desired number of elements in prefix. NB: I refer to `number of elements` as to size of the full schema of search bound, not the actual number of elements in prefix

as user of this method, I want to have convenient utility method to convert given BinaryTuple which contains only relevant fields to a proper prefix

Reply:
My question is, why would a _prefix_ of a tuple have more elements than a tuple?

Reply:
because, although they have pretty similar name, it's a complete different object. Prefix is a part of index api, it represents partially filled search bound of a particular index. BinaryTuple, on the other hand, is a general purpose container.

So we have general purpose container that we want to reuse as search bound, but we got only relevant fields in it. In that case, size of BinaryTuple will be less than full size of index-related BinaryTuplePrefix